src/D2L.Security.OAuth2/Keys/JsonWebKey.cs

```diff
@@ -104,12 +104,17 @@ public abstract class JsonWebKey {
 						throw new JsonWebKeyParseException( "RSA JSON web key has private key material" );
 					}
 
-					return new RsaJsonWebKey(
+					if( !RsaJsonWebKey.TryParse(
 						id: id,
 						expiresAt: expiresAt,
-						n: data[ "n" ].ToString(),
-						e: data[ "e" ].ToString()
-					);
+						n: data["n"].ToString(),
+						e: data["e"].ToString(),
+						rsaJsonWebKey: out RsaJsonWebKey rsaJsonWebKey
+					) ) {
+						throw new JsonWebKeyParseException( "decoded 'n' parameter starts with 0 byte in RSA JSON web key" );

```

Comment:
nit: it would probably be better for `RsaJsonWebKey` to simply throw, all things considered. the `TryX` pattern kinda has the flaw of being opaque about what caused a thing to fail, so when you do want details (e.g. to format a specific error) you need another `out` to explain what the failure was.

That being said I'm fine with this for now. We should refactor this library to not use exceptions eventually, and I could grab this one later.